### ISSUE TITLE
[Feature] Save edits made while in a mode

### ISSUE THREAD

**Issue by thimblebird** [Darkere/ConfigSwapper#3]
this would be incredibly useful; having, for example, a command like `/mode save` or `/mode overwrite` to save the changes made in the current mode.

that's all. you really saved my modpack, regardless of this feature request :)

**Comment by Darkere** [comment]
Cool idea, but unfortunately not really feasible to figure out what you have changed.

**Comment by thimblebird** [comment]
you're pretty quick with closing issues... if you made this an "experimental feature", it would be a thing of copying all current config files (e.g. `/root/config/**/*`) into the config folder of the mode folder.

this would allow for a `custom` mode folder so players can easily change configs and save their changes.

marking it as experimental with a warning of "this gives a lot of power to players and the potential for them to ruin their modes" would, in my mind, make it feasable to include without the overhead of "figuring out what changed".

let me know your thoughts :)

**Comment by thimblebird** [comment]
a `/mode none` command would possibly also work, where it doesn't use a specific mode. this would make the `root/config/` folder the custom one, as long as they don't change modes...
then again, the proposal would also make for easier adding/editing modes in the game.

**Comment by Darkere** [comment]
That would get around that issue, however... At that point you can just copy paste the folder yourself :D

**Comment by thimblebird** [comment]
it's not about me, it's about players that don't even know there are config files or where to find them. accessibility for non-tech-savvy people 🐢

**Comment by Darkere** [comment]
If you don't know where the config folder is, this is not the mod for you ;)

**Comment by thimblebird** [comment]
nobody will just install your mod and call it a day playing minecraft. isn't this a tool designed for modpack developers?

all kinds of people play modpacks; your 10 years old nephew to the 65 years old grandma. would you be so kind to elaborate how this would not be a feature that helps players AND developers?

**Comment by Darkere** [comment]
Yes. It's a tool for modpack developers. And modpack developers are very familiar with where the config folders are. 

Players don't need to go into the configs, and they certainly shouldn't be overriding the config modes causing issues with future updates.

**Comment by thimblebird** [comment]
players should be able to edit configs to tweak stuff they don't like in a mode and have those changes not get overridden on game restart... or does that not happen?

**Comment by Darkere** [comment]
It does. However if players copied over all configs it would then override config changes that could come from an update to the pack.

**Comment by thimblebird** [comment]
what about an option to lock specific modes so they can't be overwritten? (which would be on by default, as per current behaviour)
e.g. a list of mode names to allow being overwritten

**Comment by Darkere** [comment]
I'm unsure at this point what exactly you are arguing for?

**Comment by thimblebird** [comment]
players can't set their own configs via the normal config menus in-game because they are deleted when restarting the game, switching modes or after pack updates. is there a workaround i'm not seeing or what's with your question?

**Comment by Darkere** [comment]
I forget the ingame menu exists... 

I can potentially add something to that menu that allows you to overwrite that value in the current mode. Not certain. But I needed to investigate that for another mod anyways.

**Comment by thimblebird** [comment]
what do you think about an "allow list" in the `\config\configswapper.json` where one can list modes that are allowed to be customized? i could then add a "custom" mode with all mod configs inside (or leave it empty too) and a player could switch to that mode, changing everything to whatever.

configswapper would then simply overwrite the whole individual (config) file in the `custom` mode folder with the "original" (config) file that the user just change via in-game menus. no need to watch for specific values then, right?

**Comment by Darkere** [comment]
The issue I mentioned at the beginning, where I would need to monitor your changes is only a problem if you are doing these changes outside the game in a text editor.

If you are changing values ingame with some menu mod then I can potentially read the values from there.
